src/core/plugin/luapi_application.h

```diff
@@ -396,6 +396,53 @@ static int applib_sidebarAction(lua_State* L) {
     return 0;
 }
 
+/*
+ * Get the index of the currently active sidebar-page.
+ *
+ * Example: app.getSidebarPageNo() -- returns e.g. 1
+ */
+
+static int applib_getSidebarPageNo(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Sidebar* sidebar = plugin->getControl()->getSidebar();
+    lua_pushinteger(L, sidebar->getSelectedPage());
+    return 1;
+}
+
+/*
+ * Set the currently active sidebar-page by its index.
+ *
+ * Look at src/core/gui/sidebar/Sidebar.cpp to find out which index corresponds to which page (e.g. currently 0 is the
+ * page with the TOC/index if available).
+ *
+ * Example: app.setSidebarPageNo(2) -- sets the sidebar-page to preview Layer
+ */
+static int applib_setSidebarPageNo(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Sidebar* sidebar = plugin->getControl()->getSidebar();
+
+    // Discard any extra arguments passed in
+    lua_settop(L, 1);
+    if (!lua_isnumber(L, 1)) {
```

Comment:
It should be an integer
```suggestion
    if (!lua_isinteger(L, 1)) {
```